lib/reporters/base.js

```diff
@@ -333,6 +340,67 @@ Base.prototype.epilogue = function() {
 };
 
 /**
+ * Opens for writing the file referenced on the `optionName` option
+ * of the reporter options, if any.
+ * Call this method to support writing to this file when calling
+ * write or writeLine.
+ *
+ * @param {string} optionName the name of the option, defaults to 'output'.
+ * @api public
+ */
+Base.prototype.openOutput = function(optionName) {
+  var output = this.reporterOptions && this.reporterOptions[optionName || 'output'];
+  if (output) {
+    if (!fs.createWriteStream) {
+      throw new Error('file output not supported in browser');
+    }
+
+    mkdirp.sync(path.dirname(output));
+    this.fileStream = fs.createWriteStream(output);
+  }
+};
+
+/**
+ * Write to reporter output stream.
+ *
+ * @param {string} str
+ * @api public
+ */
+Base.prototype.write = function(str) {
+  if (this.fileStream) {
+    this.fileStream.write(str);
+  } else {
+    process.stdout.write(str);
+  }
+};
+
+/**
+ * Write to reporter output stream, adding an EOL at the end.
+ *
+ * @param {string} line
+ * @api public
+ */
+Base.prototype.writeLine = function(line) {
+  this.write(line + EOL);
+};
+
+/**
+ * Close the output stream and callback with failures.
+ *
+ * @param failures
+ * @param {Function} fn
+ */
+Base.prototype.done = function(failures, fn) {
+  if (this.fileStream) {
+    this.fileStream.end(function() {
+      fn(failures);
+    });
+  } else {
+    fn(failures);
```

Comment:
I think `done` already existed as such and received a failures and callback, I only added a default implementation to `base` and use the mechanism used on the previous PR to call done on each reporter. 

Since it already received a failures and a callback, I'm not sure we should change that, but I defer to you on that.